packages/labs/ssr-client/src/controllers/server-controller.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {ReactiveController} from 'lit';
+
+/**
+ * ServerController extends the Reactive Controller interface to allow
+ * asynchronous work to occur during SSR on the server.
+ */
+export interface ServerController extends ReactiveController {
+  /**
+   * `serverUpdateComplete` is accessed on the server during SSR. When
+   * implemented, SSR rendering will pause until all the controllers on this
+   * ReactiveController's host have settled.
+   */
+  serverUpdateComplete: Promise<unknown>;
```

Comment:
I think we want to say `undefined | Promise<unknown>`

e.g. Task is a controller that might or might not want to produce a promise here, based on configuration

Reply:
Should it just be optional?